test/stream.js

```diff
@@ -40,18 +40,13 @@ metatests.test(
 );
 
 metatests.test('MemoryWritable handles custom sizeLimit', async test => {
-  for (const path of fixturesPaths) {
-    const originalReadable = fs.createReadStream(path);
-    const memoryStream = new common.MemoryWritable(40);
-    originalReadable.pipe(memoryStream);
-    const result = memoryStream.getData();
-    if (path.includes('empty')) {
-      await test.resolves(result, await readFile(path));
-    } else {
-      await test.rejects(
-        result,
-        new RangeError('size limit exceeded by 20 bytes')
-      );
-    }
-  }
+  const totalSize = 60;
+  const limitSize = 40;
+  const memoryStream = new common.MemoryWritable(limitSize);
+  const result = memoryStream.getData();
+  memoryStream.write(Buffer.alloc(totalSize));
+  test.rejects(
+    result,
```

Comment:
Nit: I'd prefer to request promise in a function to `rejects`:
 ```suggestion
    () => {
      const result = memoryStream.getData();
      memoryStream.write(Buffer.alloc(totalSize));
      return result;
    },
```